test/integration/replicaset/replicaset_test.go

```diff
@@ -162,19 +173,252 @@ func rmSimpleSetup(t *testing.T) (*httptest.Server, framework.CloseFunc, clients
 	return s, closeFn, clientSet
 }
 
+// Run RS controller and informers
+func runControllerAndInformers(rm *replicaset.ReplicaSetController, informers informers.SharedInformerFactory) chan struct{} {
+	stopCh := make(chan struct{})
+	informers.Start(stopCh)
+	go rm.Run(5, stopCh)
+	return stopCh
+}
+
 // wait for the podInformer to observe the pods. Call this function before
 // running the RS controller to prevent the rc manager from creating new pods
 // rather than adopting the existing ones.
 func waitToObservePods(t *testing.T, podInformer cache.SharedIndexInformer, podNum int) {
-	if err := wait.Poll(10*time.Second, 60*time.Second, func() (bool, error) {
+	if err := wait.PollImmediate(interval, timeout, func() (bool, error) {
 		objects := podInformer.GetIndexer().List()
-		if len(objects) == podNum {
-			return true, nil
-		} else {
-			return false, nil
+		return len(objects) == podNum, nil
+	}); err != nil {
+		t.Fatalf("Error encountered when waiting for podInformer to observe the pods: %v", err)
+	}
+}
+
+func createRSsPods(t *testing.T, clientSet clientset.Interface, rss []*v1beta1.ReplicaSet, pods []*v1.Pod) ([]*v1beta1.ReplicaSet, []*v1.Pod) {
+	var ns string
```

Comment:
Perhaps we can bring in the namespace as a function parameter, saving 6 lines of codes.

Reply:
Noted, I will create `rsClient` and `podClient` in the loops.